### ISSUE TITLE
YouTube

### ISSUE THREAD

**Issue by ouija** [klattimer/LGWebOSRemote#28]
Finally got this awesome script working tonight on my QNAP NAS with Entware!

However, disappointed to find out that the openYoutubeId and openYoutubeURL functions don't seem to have any effect on my TV;  YouTube will open, but doesn't load any video, regardless of what you specify.

Using the openBrowserAt command with the URL works, but I'd love to get the YouTube app working properly.  I thought maybe I could try to get it working via the openAppWithPayload command and passing the ID as a parameter to it, but that doesn't work either.

Any ideas why this doesn't work? 

**UPDATE:**  I think this is because the linux machine running the script actually needs to "pair" with the YouTube app (similar to what a phone does when casting to it)

I found some examples of this in other sources:

- https://github.com/ConnectSDK/Connect-SDK-iOS-Core/blob/master/Services/WebOSTVService.m
- https://github.com/ConnectSDK/Connect-SDK-Android-Core/blob/master/src/com/connectsdk/service/WebOSTVService.java
- https://github.com/ConnectSDK/Connect-SDK-Android-Core/pull/93/files?diff=split&short_path=04c6e90&unchanged=expanded
- https://www.programcreek.com/java-api-examples/index.php?api=com.connectsdk.core.AppInfo

Where essentially the parameter being passed is `contentId` with some additional variables added to it:
`params.put("contentId", String.format("%s&pairingCode=%s&t=%.1f", contentId, UUID.randomUUID().toString(), startTime));`

so basically it needs a random UUID passed with a time value?

Something like:
`{"contentId": videoid+'&pairingCode='+randomUUID+'&t=1'}}`

I tried editing the script and passing something like this when openYoutubeId was being called, but it didn't help;  YouTube would still open but no video would play.

**Comment by klattimer** [comment]
The youtube app was removed from my TV in an update a while back, I have no way of testing the youtube app and even recommend using the openBrowserAt method instead in the readme. Closing this issue.
